docker/addons/vault/scripts/vault_copy_env.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/bash
+# Copyright (c) Abstract Machines
+# SPDX-License-Identifier: Apache-2.0
+
+set -euo pipefail
+
+scriptdir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+# default env file path
+env_file="docker/.env"
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --env-file)
+            if [[ "$2" = /* ]]; then
```

Comment:
@JeffMboya What does it does ?

Reply:
@arvindh123 This checks if the second argument (`$2`) is an absolute path, which in Unix/Linux means it starts with a `/`. If it is an absolute path, the script simply assigns it to the `env_file` variable as it is.

Reply:
The error happens because `/*` is a wildcard pattern that matches all directories and files at the root level, and the script is picking up `/bin` as one of those matches.

Reply:
Why do we need this logic ? Just trying to understand you code 
Particularly why we need to convert relative path to Absolute Path ? 
```bash
            if [[ "$2" = /* ]]; then
                env_file="$2"
            else
                env_file="$(realpath -m "$PWD/$2")"
            fi
            shift
            ;;
```

Reply:
> Simply we can throw error , If `$2` is empty.

@arvindh123 an error is thrown when `$2` is empty:
```sh
$ magistrala/docker/addons/vault/scripts/vault_copy_env.sh --env-file   
magistrala/docker/addons/vault/scripts/vault_copy_env.sh: line 15: $2: unbound variable
```

Reply:
Even it was fully qualified , without check it existence , we could not say that file is exists . 
Even it is relative path , if file exists is checked ,then we need to verify it is text file of env kind. (but we can skip this verification but file existence is needed)

Reply:
> Even it was fully qualified , without check it existence , we could not say that file is exists .
> Even it is relative path , if file exists is checked ,then we need to verify it is text file of env kind. (but we can skip this verification but file existence is needed)

Yep, that makes sense. The most important part is making sure the file is there. I think the extra check is nice to have but not absolutely necessary.